### PR TITLE
Add rewards params

### LINKED ISSUE #1289: Add RewardsParameters to ProtocolParameters - Rust

### PR DESCRIPTION
**Pull request by thibault-martinez**
Implements https://github.com/iotaledger/tips-draft/pull/25/commits/2afbbfb26a46ec112bfac213d234635f2fc80aca
Closes https://github.com/iotaledger/iota-sdk/issues/1289

### CODE REVIEW

**Review by Thoralf-M — DISMISSED** [review]

**Review by Alex6323 — DISMISSED** [review]

File: sdk/src/types/block/mana/rewards.rs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(rename_all = "camelCase")
+)]
+#[packable(unpack_error = Error)]
+#[getset(get_copy = "pub")]
+pub struct RewardsParameters {
+    /// The number of validation blocks that should be issued by a selected validator per slot during its epoch duties.
+    validator_blocks_per_slot: u8,
+    /// Used for shift operation for calculation of profit margin.
```
Comment (DaughterOfMars) [inline sdk/src/types/block/mana/rewards.rs]:
```suggestion
    /// Used for shift operation during calculation of profit margin.
```

File: sdk/src/types/block/mana/rewards.rs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    profit_margin_exponent: u8,
+    /// The length in epochs of the bootstrapping phase.
```
Comment (DaughterOfMars) [inline sdk/src/types/block/mana/rewards.rs]:
```suggestion
    /// The length of the bootstrapping phase in epochs.
```

File: sdk/src/types/block/mana/rewards.rs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    bootstrapping_duration: EpochIndex,
+    /// The coefficient used for calculation of initial rewards.
+    #[cfg_attr(feature = "serde", serde(with = "crate::utils::serde::string"))]
+    mana_share_coefficient: u64,
+    /// The exponent used for calculation of the initial reward.
+    decay_balancing_constant_exponent: u8,
+    /// Needs to be an integer approximation calculated based on chosen DecayBalancingConstantExponent.
```
Comment (DaughterOfMars) [inline sdk/src/types/block/mana/rewards.rs]:
```suggestion
    /// An integer approximation which is calculated using the `decay_balancing_constant_exponent`.
```
Though an overall better description is needed I think

File: sdk/src/types/block/mana/rewards.rs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    #[cfg_attr(feature = "serde", serde(with = "crate::utils::serde::string"))]
+    decay_balancing_constant: u64,
+    /// The exponent used for shifting operation in the pool rewards calculations.
```
Comment (DaughterOfMars) [inline sdk/src/types/block/mana/rewards.rs]:
```suggestion
    /// The exponent used for shifting operation during the pool rewards calculations.
```

**Review by DaughterOfMars — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**